docs/backend/setup/database-migrations.md

```diff
@@ -98,56 +98,113 @@ When you are writing the code for your new migration, there are a few things you
+    (
+        id uuid PRIMARY KEY NOT NULL,
+        created_at timestamp NOT NULL,
+        updated_at timestamp NOT NULL,
+        name text NOT NULL,
+        birthday date,
+        gotcha_day date,
+        bio text,
+        weight int
+    );
+    
+    COMMENT on TABLE cats IS 'Store cats and their details.';
+    COMMENT on COLUMN cats.name IS 'The official name of a cat...nicknames might need their own table...';
+    COMMENT on COLUMN cats.birthday IS 'Date the cat was born.';
+    COMMENT on COLUMN cats.gotcha_day IS 'Date the cat was adopted. May be same as birthday.';
+    COMMENT on COLUMN cats.bio IS 'Big text field for owners to tell you everything about their cat.';
+    COMMENT on COLUMN cats.weight IS 'Current weight of the cat, or at least what they weighed the last time they let you weigh them.';
+    ```
 
-```go
-package models
-
-import (
-  "time"
-
-  "github.com/gobuffalo/pop"
-  "github.com/gobuffalo/validate"
-  "github.com/gobuffalo/validate/validators"
-  "github.com/gofrs/uuid"
-)
-
-// NewTable represents a new table
-type NewTable struct {
-  ID        uuid.UUID `json:"id" db:"id"`
-  Column1   string    `json:"column1" db:"column1"`
-  Column2   *string   `json:"column2" db:"colunn2"`
-  CreatedAt time.Time `json:"created_at" db:"created_at"`
-  UpdatedAt time.Time `json:"updated_at" db:"updated_at"`
-}
+    Some things to note:
+
+    1. The SQL keywords don't need to be capitalized, but it can be nice to differentiate some of them from certain 
+       things like the field names or types.
+    1. We want to add comments for every column other than the `id`, `created_at`, and `updated_at` fields. This makes 
+       it easier for future folks to understand what the columns were meant for.
+    1. You might see some migrations use `varchar` instead of `text`. It's the same thing in the end, see
+       [Postgresql Character Types](https://www.postgresqltutorial.com/postgresql-char-varchar-text/) for more info.
+    
+1. Now you can create a new file in `pkg/models/` named after your new model. So for this, we'll have a new file 
+    `pkg/models/cat.go` that will look something like this:
+
+    ```go
+    package models
+    
+    import (
+        "time"
+    
+        "github.com/gofrs/uuid"
+    
+        "github.com/transcom/mymove/pkg/unit"
+    )
+    
+    // Cat contains all the information relevant to a cat...
+    type Cat struct {
+        ID         uuid.UUID   `json:"id" db:"id"`
+        CreatedAt  time.Time   `json:"created_at" db:"created_at"`
+        UpdatedAt  time.Time   `json:"updated_at" db:"updated_at"`
+        Name       string      `json:"name" db:"name"`
+        Birthday   *time.Time  `json:"birthday" db:"birthday"`
+        GotchaDay  *time.Time  `json:"gotcha_day" db:"gotcha_day"`
+        Bio        *string     `json:"bio" db:"bio"`
+        Weight     *unit.Pound `json:"weight" db:"weight"`
+    }
+    
+    // Cats is a list of Cats
+    type Cats []Cat
```

Comment:
To be honest, I don't and haven't had to really work with them. Do you have an example in particular? I'm thinking that at least for when it's a return type, declaring its name in the function definition could make it easier to work with since you could append to the return var directly without having to initialize it yourself. I guess for parameters I wonder if there's a need to declare the variable being worked with as a slice of the model directly instead of using the plural type, which is why I assume there's a need for type conversions.   

cc: @gidjin, you mentioned you liked this pattern, not sure if you know more pros/cons.

Reply:
If I gave the impression I liked the pattern sorry for that. I'm indifferent too it tbh. My desire is for us to be consistent so either we have this pattern everywhere or we don't. As it stands we have it in some places but not others. So for this doc I'd like it to land on the recommendation so we have a place to point and steer PRs to consistency. 

I don't really see value of `Cats` over `[]Cat` both tell me it's a list (slice in go here I believe) of `Cat` models, and I can see an argument that the latter is more explicit about it.

Reply:
I will add that saying we should have plural types is easier than saying we shouldn't. Since the former means we just have to add the type if it's missing and nothing breaks. But the latter will require replacing all the `Cats` with `[]Cat` which isn't hard, just might be a lot of places. And one could argue that when going the other direction you should make the same search and replace, but if you don't the code still works. Hopefully this has been clear.

Reply:
I'm okay with us being consistent and keeping with the types because they are used in all of the has_many associations. My main point was just that this has tripped me up a bunch and I haven't come across a situation where this is a clear win.  I don't see Pop recommending them or anything https://www.gobuffalo.io/en/docs/db/relations/onetomany/#example